pkg/resourcesets/collector.go

```diff
@@ -235,7 +235,7 @@ func (h *ResourceHandler) filterByNameAndLabel(ctx context.Context, dr dynamic.R
 		// no filters for names of the resource, return all objects obtained from the list call
 		return resourceObjectsList.Items, nil
 	}
-	// filter out using ResourceNameRegexp
+	// filter out using ResourceNameRegexp and ExcludeResourceNameRegexp
 	if filter.ResourceNameRegexp != "" {
```

Comment:
The problem with this solution is that if `ResourceNameRegexp` is empty (`""`) then it will never hit the ExcludeResourceNameRegexp logic. I think the logic for exclude needs to be separate.